### ISSUE TITLE
scheduler: more flexible and more unified scatter-region-scheduler

### ISSUE THREAD

**Issue by lhy1024** [tikv/pd#5603]
### Describe your feature request-related problem

We try to add scatter for a table with nine indexes, which makes PD CPU increase to 2000%.

Because it will add n+1 scatter-range-scheduler when we add scatter to a table with n indexes and every scatter-range-scheduler 
 contains balance-leader-scheduler and balance-region-scheduler in the range cluster which specified the start key and end key.

### Describe the feature you'd like

we need a more flexible and more unified scatter-region-scheduler. 

### Describe alternatives you've considered

Flexible means we can choose any index or table. And it can run steadily.

Unified means we can add all scatter to the balance-leader-scheduler and balance-region-scheduler, which doesn't cost external threads and can reduce conflict with the balance-leader-scheduler and balance-region-scheduler. Of course, we also need to create operators batch to avoid being too slow.

**Comment by lhy1024** [comment]
It is related with #4272

**Comment by yzhan1** [comment]
Hi @lhy1024 @bufferflies we have seen this issue in our production as well. Is there a timeline/plan to fix this?

**Comment by bufferflies** [comment]
> Hi [@lhy1024](https://github.com/lhy1024) [@bufferflies](https://github.com/bufferflies) we have seen this issue in our production as well. Is there a timeline/plan to fix this?

You can see the https://github.com/tikv/pd/issues/4272 can satisfy your meets

**Comment by yzhan1** [comment]
@bufferflies #4272 is still open. I assume then both issues are not fixed yet?

**Comment by bufferflies** [comment]
#4272 function part has been finished, this feature needs to backport to the release-8.5

**Comment by yzhan1** [comment]
@bufferflies could you share the PR that fixes it? I don't see it getting linked in the issue.

**Comment by yzhan1** [comment]
@bufferflies Maybe I misunderstood. Are you saying that #8986 will fix all existing issues of scatter-range by introducing the new balance-range scheduler?

**Comment by bufferflies** [comment]
> [@bufferflies](https://github.com/bufferflies) Maybe I misunderstood. Are you saying that [#8986](https://github.com/tikv/pd/issues/8986) will fix all existing issues of scatter-range by introducing the new balance-range scheduler?

Yes, I suggest using the scatter scheduler if the table is small( the count of partitions and indices is small).
